### PR TITLE
Update ColonyNetwork to support v5 locally

### PR DESCRIPTION
**Pull request by rdig**
This PR updates the `feature/simplified-colony` with the latest `colonyNetwork` developments _(`develop` branch)_ in order for use the use all the new features that the Network team are pushing.

As such, this branch
- [x] updates `colonyNetwork` with the latest changes from `develop`, but through our branch `lwss-dapp` _(the Network guys are trying to make this process easier in the future. See: JoinColony/colonyNetwork#902)_
- [x] updates `colonyJS` to the latest `3.0.0-alpha.6` release
- [x] Adjusts the `OneTxPayment`, `colonyManager` and `colonyCreate` saga in order for the colony creation process to still function properly. Note that all these changes were already done by @chmanie in #2260, I've just cherry-picked them so as to not cause merge conflicts:
  - ee3743c15314164a89083c65f0ab68b40d0a1884
  - 2d68aead5394038d4712209bbbb0061d96346fb6
  - b6193b8e0266bc017750f29a10c1bfadbbc7a799

### **Notes for Reviewers**

**Please test this locally:**
- remove your modules
- reinstall them
- run provision _(check if a different `yarn.lock` file is generated inside `src/lib/colonyNetwork`, as that's not good)_
- start the dev environment
- create a user
- create a colony

### CODE REVIEW

File: src/lib/ColonyManager/ColonyManager.ts [hunk 1]
```diff
@@ -121,23 +136,23 @@ export default class ColonyManager {
         return this.networkClient;
       }
       case ClientType.ColonyClient: {
+        if (!identifier)
```
Comment (alicjakujawa) [inline src/lib/ColonyManager/ColonyManager.ts:139]: shouldn't we just make it required if this is required for all clients? 🤔

Comment (rdig) [reply src/lib/ColonyManager/ColonyManager.ts:139]: I'm not sure. This came from one of the commits I cherry-picked from #2260, commit 2d68aead5394038d4712209bbbb0061d96346fb6

I guess I'll have to look into it :+1:

Comment (chmanie) [reply src/lib/ColonyManager/ColonyManager.ts:139]: The `NetworkClient` does not require (and cannot have) an identifier, yet also can be returned from this function. This is here to make typescript happy.

**Review by alicjakujawa — APPROVED** [review]
🔥

**Pull request merged**